### ISSUE TITLE
Reason for use of ImmutablePropTypes?

### ISSUE THREAD

**Issue by markerikson** [shinzui/react-semantic-ui-kit#33]
I see that ImmutablePropTypes is specified as a dependency, and Immutable.js as a dev dependency.  It looks like the only use of IMT is for the "errors" prop in FormField, which does look like it's being used expecting an actual Immutable.List.  

Is there a specific reason why it's being used there?  It doesn't seem like there's any actual real benefit for introducing this dependency just for that one field.

**Comment by shinzui** [comment]
@markerikson You are right that it should not be declared as a dependency. I started this project because I needed to share my form handling components across projects, and I extracted the FormField first from a project that heavily uses Immutable.js.

I'll remove the dependency this weekend and close this issue.

**Comment by markerikson** [comment]
Cool.  Just playing around with this for the first time.  Definitely like what I see so far.

Thanks!

**Comment by markerikson** [comment]
So I'm actively using React-Semantic-UI-Kit in my prototype, and liking it so far.  Unfortunately, besides the mostly-unnecessary Immutable dependency, I'm now actually running into this specific prop being a problem, as I'm trying to mark a FormField as having an error.  Also, the FormField doesn't accept any additional classnames to be applied to the field, so I can't even manually apply the "error" class that Semantic-UI's CSS uses.

**Comment by shinzui** [comment]
Sorry for not updating this. I've been busy at work, and I am in the middle of a move. I'll resume working on this very soon. 

I did not remove the dependency yet because I want to think about the API. Let me know if you have any suggestions. I have a few ideas, I just need to block some time to implement them.

**Comment by markerikson** [comment]
What, Real Life interfering?  Un-possible! :)

API-wise, the pain points I've noticed so far with the lib are:
- Wanting to attach arbitrary classnames to a given component's element in addition to whatever it's generating based on props
- Menus filtering out any child that's not a direct MenuItem ( #35 )
- Form components complaining if a child is not specifically a FormField (even if the component happens to actually render a FormField internally)
- The remnants of the Immutable.js dependency

That said, the abstraction layers are pretty nice overall, and definitely simpler than trying to always add the classes by hand.

I may put together a PR or two if I have some time.

**Comment by shinzui** [comment]
I am planning to address the last 3 items. As far as supporting arbitrary class names, most components support `className` with the exception of `FormField`. I'll quickly add support for that right now.

**Comment by markerikson** [comment]
Oh, one other one: right now I've got several buttons in a form that are auto-submitting the whole form on click and are causing the page the page to reload unless I specifically call `e.preventDefault(); e.stopPropagation()`. Googling/SO says the answer to that is to mark the button as `<button type="button">` ( http://stackoverflow.com/questions/9824808/disable-form-auto-submit-on-button-click ), but the Button component doesn't support that atm.  Similarly, I'd like to disable autocomplete on my forms, but that also requires an attribute on the form tag itself ( https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion ).

If some of the components supported passing `{...otherProps}` to the main tags, that would presumably take care of the scenarios.

Thanks for the lib and the response!

**Comment by shinzui** [comment]
@markerikson could you please open 3 separate issues. One for the passthrough properties, and one for each of the other two since they should just work.

**Comment by markerikson** [comment]
Sure. "Other two" being menu filtering and form field children?

**Comment by markerikson** [comment]
Erm. Lemme step back through everything I've mentioned:

- Immutable.js dependency: this issue
- Arbitrary additional classnames: you said you'll take care of it
- Passthrough props: needs an issue
- Menu filtering: #35 
- FormFieldGroup complaining about non-FormField children: needs an issue

So, new issue for passthrough props, new issue for FormFieldGroup, and which was the third you were referring to?

**Comment by shinzui** [comment]
I've already pushed support for `className` on the `FormField` component. Let me know if you need it anywhere else, it's already supported on most components. 

"auto-submitting the whole form on click" should be an issue.
